src/main/java/com/dsrm/dsrmbackend/controllers/SecurityConfig.java

```diff
@@ -0,0 +1,45 @@
+package com.dsrm.dsrmbackend.controllers;
+
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.security.config.annotation.authentication.builders.AuthenticationManagerBuilder;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.security.crypto.password.PasswordEncoder;
+
+@Configuration
+@EnableWebSecurity
+public class SecurityConfig extends WebSecurityConfigurerAdapter {
+
+    @Override
+    protected void configure(HttpSecurity http) throws Exception {
+        http.authorizeRequests()
```

Comment:
It should be something like that.
```
    @Bean
    public SecurityFilterChain filterChain(HttpSecurity http) throws Exception {
        http
                .cors()
                .disable()
                .csrf()
                .disable()
                .authorizeHttpRequests()
                .mvcMatchers(HttpMethod.OPTIONS, "/**").permitAll()
                .mvcMatchers("/v3/api-docs/**", "/configuration/**", "/swagger*/**", "/webjars/**", "/actuator/**", "/authenticate").permitAll()
                .mvcMatchers("/admin/**").hasAuthority("ADMIN")
                .anyRequest().hasAuthority("USER")
                .and()
                .oauth2ResourceServer()
                .jwt()
                .decoder(jwtLocalDecoder)
                .jwtAuthenticationConverter(jwtAuthenticationConverter)
                .and()
                .and();
        return http.build();
    }
```
Aparat of that we need to implement missing jwtdecoder and jwtAuthenticationConverter.